src/encoded/static/components/filegallery.js

```diff
@@ -3256,26 +3350,97 @@ FileGalleryRendererComponent.contextTypes = {
 const FileGalleryRenderer = auditDecor(FileGalleryRendererComponent);
 
 
-const CollapsingTitle = (props) => {
-    const { title, handleCollapse, collapsed } = props;
+const CollapsingTitle = (props, reactContext) => {
+    const { title, handleCollapse, collapsed, context, filters, analysisObjectKey, outputCategory, outputType, totalFiles, isDownloadable } = props;
+
+    const [downloadModalVisibility, setDownloadModalVisibility] = useState(false);
+
+    const handleDownloadClick = () => {
+        const { accession } = context;
+        const type = context && context['@type'] ? context['@type'][0] : 'Experiment';
+        let analysisObjectId = '';
+
+        if (context.analysis_objects) {
+            const analysisObjects = context.analysis_objects.find((analysisObject) => analysisObject.accession === analysisObjectKey);
+
+            if (analysisObjects) {
+                analysisObjectId = analysisObjects['@id'];
+            }
+        }
+
+        const fileTypes = filters.file_type && filters.file_type.length > 0
+            ? filters.file_type.map((fileType) => `&files.file_type=${encodedURIComponent(fileType)}`).join('')
+            : '';
+        const biologicalReplicates = filters.biological_replicates && filters.biological_replicates.length > 0
+            ? filters.biological_replicates.map((replicate) => `&files.biological_replicates=${encodedURIComponent(replicate)}`).join('')
+            : '';
+        const outputTypes = filters.output_type && filters.output_type.length > 0
+            ? filters.output_type.map((o) => `&files.output_type=${encodedURIComponent(o)}`).join('')
+            : '';
+        const fileStatus = document.querySelector('[name="filterIncArchive"]').checked
+            ? ''
+            : '&files.status=released';
+        const outputCategoryQueryString = outputCategory ? `&files.output_category=${outputCategory}` : '';
+        const outputTypeQueryString = outputType ? `&files.output_type=${outputType}` : '';
+
+        if (outputCategoryQueryString || outputType) {
+            reactContext.navigate(`/batch_download/?type=${type}&accession=${accession}${outputCategoryQueryString}${outputTypeQueryString}${fileTypes}${outputTypes}${biologicalReplicates}${fileStatus}`);
+        } else {
+            reactContext.navigate(`/batch_download/?type=Analysis&@id=${analysisObjectId}&${fileTypes}${outputTypes}${biologicalReplicates}${fileStatus}`);
+        }
+        setDownloadModalVisibility(false);
+    };
```

Comment:
Seems like a very large function to get re-parsed on each render (not that we have a specific criterium for that). Could you put it in the global space and pass in `context`, `filters`, `outputCategory`, `outputType`, and `inclusionOn`?

Reply:
This was not changed as although there is common code with another function, it is not common enough to warrant a function that is passed. The function is also not so large or resource-heavy enough to warrant a function of its own either. Let me know if you now beg to differ.

Reply:
Agreed.